@stellar/anchor-tests/src/tests/sep31And38/transactions.ts

```diff
@@ -0,0 +1,275 @@
+import { Request } from "node-fetch";
+import { validate } from "jsonschema";
+import { Keypair, Memo } from "stellar-sdk";
+
+import { Test, Result, Config, NetworkCall } from "../../types";
+import { hasDirectPaymentServer } from "../sep31/toml";
+import { differentMemosSameAccount } from "../sep12/putCustomer";
+import { canCreateQuote } from "../sep38/postQuote";
+import { genericFailures, makeFailure } from "../../helpers/failure";
+import { makeRequest } from "../../helpers/request";
+import {
+  getTransactionSchema,
+  postTransactionsSchema,
+} from "../../schemas/sep31";
+import { hasExpectedAssetEnabled } from "../sep31/info";
+import { getQuotesSupportedFromInfo } from "../../helpers/sep31";
+
+const postTransactionsGroup = "POST /transactions";
+const getTransactionsGroup = "GET /transactions/:id";
+const tests: Test[] = [];
+
+let quotesSupported: boolean;
+
+const canCreateTransaction: Test = {
+  assertion: "[with SEP-38, quotes_required]  can create a transaction",
+  sep: 31,
+  group: postTransactionsGroup,
+  dependencies: [
+    hasDirectPaymentServer,
+    hasExpectedAssetEnabled,
+    differentMemosSameAccount,
+    canCreateQuote,
+  ],
+  context: {
+    expects: {
+      directPaymentServerUrl: undefined,
+      sendingAnchorClientKeypair: undefined,
+      sendingAnchorToken: undefined,
+      sendingCustomerId: undefined,
+      sep38QuoteResponseObj: undefined,
+      receivingCustomerId: undefined,
+      sep31InfoObj: undefined,
+    },
+    provides: {
+      transactionId: undefined,
+    },
+  },
+  failureModes: {
+    INVALID_SCHEMA: {
+      name: "invalid schema",
+      text(args: any): string {
+        return (
+          "The response body from POST /transactions does not match " +
+          "the schema defined by the protocol. " +
+          "Errors:\n\n" +
+          `${args.errors}`
+        );
+      },
+      links: {
+        "POST Transaction Schema":
+          "https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0031.md#post-transactions",
+      },
+    },
+    ...genericFailures,
+  },
+  async run(config: Config): Promise<Result> {
+    const result: Result = { networkCalls: [] };
+    if (!config.sepConfig || !config.sepConfig["31"]) {
+      // this configuration is checked prior to running tests
+      // but to satisfy TypeScript we make these checks here.
+      throw "improperly configured";
+    }
+
+    let assetCode = config.assetCode;
+    if (assetCode === undefined) {
+      throw "asset not configured";
+    }
+    quotesSupported = getQuotesSupportedFromInfo(
+      this.context.expects.sep31InfoObj,
+      assetCode,
+    );
+
+    // If quotes are not supported, ignore this test, this will be addressed in SEP 31 tests
+    if (!quotesSupported) {
+      this.context.provides.transactionId = null;
+      return result;
+    }
+
+    const splitAsset =
+      this.context.expects.sep38QuoteResponseObj.sell_asset.split(":", 3);
```

Comment:
Are we verifying the format of this is correct before using it?

Reply:
Yes, this has been tested.